src/Shouldly/ShouldlyExtensionMethods/ShouldBeEnumerableTestExtensions.cs

```diff
@@ -43,6 +43,38 @@ public static void ShouldNotContain<T>(this IEnumerable<T> actual, T expected, [
                 throw new ShouldAssertException(new ExpectedActualShouldlyMessage(expected, actual, customMessage).ToString());
         }
 
+        public static void ShouldBeginWith<T>(this IEnumerable<T> actual, T expected)
+        {
+            ShouldBeginWith(actual, expected, () => null);
+        }
+
+        public static void ShouldBeginWith<T>(this IEnumerable<T> actual, T expected, string customMessage)
+        {
+            ShouldBeginWith(actual, expected, () => customMessage);
+        }
+
+        public static void ShouldBeginWith<T>(this IEnumerable<T> actual, T expected, [InstantHandle] Func<string> customMessage)
+        {
+            if (!actual.First().Equals(expected))
+                throw new ShouldAssertException(new ExpectedActualShouldlyMessage(expected, actual, customMessage).ToString());
+        }
```

Comment:
.NET API uses `StartsWith` so perhaps we should follow the heuristics and rename the methods to `ShouldStartWith`